### PR TITLE
Replace type of Mutation Field to Bool

### LINKED ISSUE #7748: OPA enable mutation causing error
**User Story**
OPA enable mutation flag `--enable-mutation`  causing User Controller Manager to crash as the argument type in the code is taken as Address not as Bool

**Acceptance criteria**
Argument Value is taken as bool in User Controller Manager so that the value remains static

### PR DESCRIPTION
**Pull request by imharshita**
Signed-off-by: Harshita <harshita.sharma6174@gmail.com>

**What this PR does / why we need it**:
Fixes issue caused by Address value of Bool Argument `--enable-mutation`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #7748

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

### CODE REVIEW

File: pkg/resources/usercluster/deployment.go [hunk 1]
```diff
@@ -171,7 +171,7 @@ func DeploymentCreator(data userclusterControllerData) reconciling.NamedDeployme
 			}
 
 			if data.Cluster().Spec.OPAIntegration != nil && data.Cluster().Spec.OPAIntegration.ExperimentalEnableMutation {
-				args = append(args, "--enable-mutation", fmt.Sprint(&data.Cluster().Spec.OPAIntegration.ExperimentalEnableMutation))
+				args = append(args, "-enable-mutation", fmt.Sprint(data.Cluster().Spec.OPAIntegration.ExperimentalEnableMutation))
```
Comment (mfranczy) [inline pkg/resources/usercluster/deployment.go]: https://pkg.go.dev/flag#hdr-Command_line_flag_syntax

```
-flag x  // non-boolean flags only
```

please change to 
```
fmt.Sprintf("-enable-mutation=%t", data.Cluster().Spec.OPAIntegration.ExperimentalEnableMutation)),
```

Comment (mfranczy) [reply pkg/resources/usercluster/deployment.go]: btw. should not be the condition the same as here [link](https://github.com/kubermatic/kubermatic/pull/7749/files#diff-fc8f73fda439bfb46949c7196ce280cfe5393c27b629249b8d342a5ff4f981f3R173)?

If OPAIntegration is disabled but experimental mutation is enabled, do we still want to run the mutation webhook? If yes, why?

**Review by mfranczy — CHANGES_REQUESTED** [review]

File: pkg/resources/usercluster/deployment.go (hunk 1 above)
Comment (imharshita) [reply pkg/resources/usercluster/deployment.go]: In that case we don't want mutation webhook, updated the changes.
fyi create a PR to remove mutation webhook when `enable-mutation:false`

**Review by mfranczy — APPROVED** [review]
/approve
/lgtm

**Pull request merged**